### PR TITLE
Add visualization launcher using hook plugin

### LINKED ISSUE #1118: Start ertviz from ERT
Add a feature where ertviz becomes the plotting solution showed when clicking `Create pliots`. This should be enabled through an optional flag similar to `--enable-new-storage` for instance `--enable-ertviz`.

It has been suggested that ertviz would be opened in a browser and not a QT widget, but that is up for debate. 

Ertviz will not work unless the storage server is also up and running. This should be checked when feature is enabled.

### PR DESCRIPTION
**Pull request by ManInFez**
resolves #1118

### CODE REVIEW

**Comment by sondreso** [comment]
Is having the users be able to type `ert --vis` vs. `webviz-ert` in their terminal enough of a justification for adding a new hook to the plugin system?

**Comment by ManInFez** [comment]
As the visualization should be integrated into ERT it makes sense to have it as a hook and not a separate entrypoint, no?

**Comment by sondreso** [comment]
If it is to be integrated, then we can list `webviz-ert` as a dependency of ERT and then provide a launch function in that module which we trigger from the entrypoint. To me it makes sense to have a plugin system for visualization tools if you could register several of them, and then select from that list and launch them. The current hook specification does not really allow for extension in the future :slightly_smiling_face:

File: ert_shared/plugins/__init__.py [hunk 1]
```diff
@@ -1 +1,13 @@
...
+
+def launch_visualization_plugin(args):
+    pm = ErtPluginManager()
+    handler = VisualizationPluginHandler()
+    pm.hook.register_visualization_plugin(handler=handler)
+    try:
+        vis_plugin = handler.get_plugin(args.name)
+        vis_plugin.run()
+    except UserWarning as e:
+        raise SystemExit(str(e))
```
Comment (sondreso) [inline ert_shared/plugins/__init__.py:13]: Maybe just have `get_plugin` return `None` in the case of no matching plugin name, and then run/exit based on that?

File: ert_shared/plugins/__init__.py [hunk 2]
```diff
@@ -1 +1,13 @@
...
+from .visualization_plugin_handler import VisualizationPluginHandler
+
+
+def launch_visualization_plugin(args):
+    pm = ErtPluginManager()
+    handler = VisualizationPluginHandler()
+    pm.hook.register_visualization_plugin(handler=handler)
+    try:
+        vis_plugin = handler.get_plugin(args.name)
+        vis_plugin.run()
```
Comment (sondreso) [inline ert_shared/plugins/__init__.py:11]: Will there maybe be a future need to forward args?

File: ert_shared/plugins/hook_specifications/visualization.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from ert_shared.plugins.plugin_manager import hook_specification
+
+
+@hook_specification
+def register_visualization_plugin(handler):
+    """
+    launch visualization solution for ERT
+    """
```
Comment (sondreso) [inline ert_shared/plugins/hook_specifications/visualization.py:8]: Could you describe the expected format of the response eg. that we expect a class with name and run() to be added using the add_plugin?

**Comment by sondreso** [comment]
This this approach looks much better!

File: ert_shared/plugins/hook_specifications/visualization.py (hunk 3 above)
Comment (ManInFez) [reply ert_shared/plugins/hook_specifications/visualization.py:8]: :+1:

File: ert_shared/plugins/__init__.py (hunk 1 above)
Comment (ManInFez) [reply ert_shared/plugins/__init__.py:13]: I think the plugin handler should blow up early if you try to use a plugin that is not installed :shrug:

Comment (sondreso) [reply ert_shared/plugins/__init__.py:13]: Alright! But let's choose something else than UserWarning

**Comment by sondreso** [comment]
Would also like some tests :grimacing:

File: docs/rst/manual/experimental/plugin_system.rst [hunk 4]
```diff
@@ -161,3 +161,37 @@ The configuration object and properties is documented in: :class:`ert_shared.plu
...
+
+.. code-block:: python
+
+
+   from ert_shared.plugins.plugin_manager import hook_implementation
+
+   class VisPlugin:
+      name = "HelloWorld"
+      def run()
+         print("Hello World")
```
Comment (sondreso) [inline docs/rst/manual/experimental/plugin_system.rst:179]:
```suggestion
   class VisPlugin:
      name = "HelloWorld"
      @staticmethod
      def run()
         print("Hello World")
```

**Review by sondreso — APPROVED** [review]
Except that small comment, this LGTM!

File: docs/rst/manual/experimental/plugin_system.rst (hunk 4 above)
Comment (ManInFez) [reply docs/rst/manual/experimental/plugin_system.rst:179]: :+1:  I fix ;)

**Pull request merged**